src/main/java/input/HandleInput.java

```diff
@@ -0,0 +1,164 @@
+                System.out.println("Todo should not be empty!");
+            }
+        } else if (input.contains("deadline")) {
+            try {
+                createDeadline(input);
+            } catch (MissingDeadlineException e) {
+                System.out.println("Include when this deadline is due as such:" +
+                        "{deadline} /by {deadline}");
+            } catch (EmptyTaskException e){
+                System.out.println("Deadline should not be empty!");
+            }
+        } else if (input.contains("event")) {
+            try {
+                createEvent(input);
+            } catch (MissingDeadlineException e) {
+                System.out.println("Include when this event ends as such:" +
+                        "{event} /from {start date} /to {end date}");
+            } catch (MissingStartException e) {
+                System.out.println("Include when this event starts as such:" +
+                        "{event} /from {start date} /to {end date}");
+            } catch (EmptyTaskException e) {
+                System.out.println("Event should not be empty!");
+            }
+        } else if (input.contains("bye")) {
+            sayBye();
+        } else {
+            throw new UnknownInputException();
+        }
+    }
+
+    public static int getIndexToMark(String input) {
+        int idx = Integer.parseInt(input.substring(input.indexOf(" ") + 1));
+        return idx - 1;
+    }
+
+    public static void listItems() {
+        System.out.println("\t Here are the tasks in your list:");
+        for (int x = 0; x < count; x++) {
+            System.out.println("\t  " + (x + 1) + "." + tasks[x].toString());
+        }
+    }
+
+    public static void unmarkItem(String input) {
+        int idx = getIndexToMark(input);
+        tasks[idx].setDone(false);
+        System.out.println("\t OK, I've marked this task as not done yet:");
+        System.out.println("\t " + tasks[idx].toString());
+    }
+
+    public static void markItem(String input) {
+        int idx = getIndexToMark(input);
+        tasks[idx].setDone(true);
+        System.out.println("\t Nice! I've marked this task as done:");
+        System.out.println("\t " + tasks[idx].toString());
+    }
+
+    public static void chat() {
+        String input;
+        do {
+            input = new Scanner(System.in).nextLine();
+            System.out.println(line);
+            if (input.equals("list")) {
+                listItems();
+            } else if (input.contains("unmark")) {
+                unmarkItem(input);
+            } else if (input.contains("mark")) {
+                markItem(input);
+            } else {
+                try {
+                    handleInput(input);
+                } catch (UnknownInputException e){
+                    System.out.println("\t I've not seen this input before. Please tell me something else I can help you with.");
+                }
+            }
```

Comment:
I think this section could be better [SLAP](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard)ed. But otherwise, the level of abstraction for this function is good!